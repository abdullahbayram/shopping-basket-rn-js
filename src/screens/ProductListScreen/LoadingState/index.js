import React from 'react';
import { View } from 'react-native';
import { ActivityIndicator } from '@components/atoms';
import PropTypes from 'prop-types';
import globalStyles from '../../../global.style';

const LoadingState = ({ color }) => (
  <View style={[globalStyles.flex, globalStyles.centerContent]}>
    <ActivityIndicator size="large" color={color} />
  </View>
);

export default LoadingState;

LoadingState.propTypes = {
  color: PropTypes.string,
};
