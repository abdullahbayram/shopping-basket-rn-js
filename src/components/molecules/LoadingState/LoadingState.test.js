import React from 'react';
import { render } from '@testing-library/react-native';
import LoadingState from '.';

describe('LoadingState Component', () => {
  it('matches the snapshot', () => {
    const { toJSON } = render(<LoadingState color="blue" />);
    expect(toJSON()).toMatchSnapshot();
  });
});
