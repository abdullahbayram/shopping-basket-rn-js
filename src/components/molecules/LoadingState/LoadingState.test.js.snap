// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingState Component matches the snapshot 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
  testID="loading-state"
>
  <ActivityIndicator
    color="blue"
    size="large"
  />
</View>
`;
